src/oneD/StFlow.cpp

```diff
@@ -517,14 +481,99 @@ void StFlow::evalResidual(double* x, double* rsd, int* diag,
                 rsd[index(c_offset_T, j)] = T(x,j) - T_fixed(j);
                 diag[index(c_offset_T, j)] = 0;
             }
+        }
+    }
+}
 
-            if (m_usesLambda) {
-                rsd[index(c_offset_L, j)] = lambda(x, j) - lambda(x, j - 1);
-            } else {
-                rsd[index(c_offset_L, j)] = lambda(x, j);
+void StFlow::evalSpecies(double* x, double* rsd, int* diag,
+                         double rdt, size_t jmin, size_t jmax)
+{
+    for (size_t j = jmin; j <= jmax; j++) {
+        if (j == 0) { // left boundary
+            double sum = 0.0;
+            for (size_t k = 0; k < m_nsp; k++) {
+                sum += Y(x,k,0);
+                rsd[index(c_offset_Y + k, 0)] =
+                    -(m_flux(k,0) + rho_u(x,0)* Y(x,k,0));
             }
-            diag[index(c_offset_L, j)] = 0;
+            rsd[index(c_offset_Y + leftExcessSpecies(), 0)] = 1.0 - sum;
+        } else if (j == m_points - 1) { // right boundary
+            double sum = 0.0;
+            for (size_t k = 0; k < m_nsp; k++) {
+                sum += Y(x,k,j);
+                rsd[index(k+c_offset_Y,j)] = m_flux(k,j-1) + rho_u(x,j)*Y(x,k,j);
+            }
+            rsd[index(c_offset_Y + rightExcessSpecies(), j)] = 1.0 - sum;
+            diag[index(c_offset_Y + rightExcessSpecies(), j)] = 0;
+        } else { // interior points
+            for (size_t k = 0; k < m_nsp; k++) {
+                double convec = rho_u(x,j)*dYdz(x,k,j);
+                double diffus = 2.0*(m_flux(k,j) - m_flux(k,j-1))
+                                / (z(j+1) - z(j-1));
+                rsd[index(c_offset_Y + k, j)]
+                = (m_wt[k]*(wdot(k,j))
+                   - convec - diffus)/m_rho[j]
+                  - rdt*(Y(x,k,j) - Y_prev(k,j));
+                diag[index(c_offset_Y + k, j)] = 1;
+            }
+        }
+    }
+}
+
+void StFlow::evalElectricField(double* x, double* rsd, int* diag,
+                               double rdt, size_t jmin, size_t jmax)
+{
+    for (size_t j = jmin; j <= jmax; j++) {
+        // The same value is used for left/right/interior points
+        rsd[index(c_offset_E, j)] = x[index(c_offset_E, j)];
+    }
+}
+
+void StFlow::computeRadiation(double* x, size_t jmin, size_t jmax)
```

Comment:
Please move before `evalContinuity`. Then the relative position still corresponds to where this code used to be.